src/content/blog/2025-03-25-tus-node-server-v2.0.0.md

```diff
@@ -0,0 +1,185 @@
+
+- Support for [custom lockers][locker] and distributed locks. This prevents
+  concurrent access to the same resources, allowing for horizontal scaling
+  without sticky sessions.
+- Configuring a [max size][]. This makes it possible to limit the maximum upload
+  size and could also help to prevent large payload attacks.
+- Supporting more [protocol extensions][extensions].
+- A new [`@tus/azure-store`][] package for
+  [Azure Blob Storage](https://azure.microsoft.com/en-us/products/storage/blobs).
+- Support for multi-tenant environments.
+- Lots of stability improvements to [`@tus/s3-store`][].
+- Allowing hooks to override metadata.
+- Customizing upload URLs and
+  [custom nested directories](https://github.com/tus/tus-node-server/tree/main/packages/server#example-store-files-in-custom-nested-directories)
+- [And much more...](https://github.com/tus/tus-node-server/compare/%40tus/server%401.0.0...%40tus/server%402.0.0)
+
+All while maintaining backwards compatibility. It has been battle tested at
+scale in multiple large companies, including in a multi-tenant setup at
+[Supabase](https://supabase.com).
+
+However, these days people run JavaScript servers in lots of places and
+environments. It has become quite common to not run your own Node.js server but
+leverage new (serverless) runtimes, such as AWS Lambda, Cloudflare, Deno Deploy,
+or Bun. Thanks to the [WinterTC](https://wintertc.org/), the Technical Committee
+on Web-interoperable Server Runtimes, we can more or less everywhere write
+servers with the web APIs: `Request` and `Response`.
+
+Or you might be using a full-stack meta-framework, such as Next.js, Nuxt, React
+Router, SvelteKit, etc. In which case you write your API or server logic there,
+with similar `Request`/`Response` request handlers.
+
+When a server is written specifically for Node.js with `http.IncomingMessage`
+and `http.ServerResponse` handlers, there is no way to integrate in those
+environments and frameworks.
+
+## tus Node.js 2.0.0
+
+### ESM-only
+
+Considered impossible for as long as most can remember, all LTS releases of
+Node.js can now
+[`require(esm)`](https://joyeecheung.github.io/blog/2024/03/18/require-esm-in-node-js/)
+from CommonJS. An incredible achievement and finally relieving maintainers from
+choosing whether to publish CommonJS, ESM, or both.
+
+With this release we bump the minimum required Node.js version from >=16
+to >=20.19.0, the specific release that backported `require(esm)`.
+
+While we don’t depend on any ESM-only packages yet, it does give us the
+flexibility to do so should we need it, ensuring we can pick the best tools to
+offer the best uploading experience.
+
+### Running anywhere where JavaScript runs
+
+tus Node.js 2.0.0 can now run in all meta frameworks (such as Next.js, Nuxt,
+React Router, SvelteKit, etc) and all Node.js compatible runtime environments
+(AWS Lambda, Cloudflare (as long as the store does not depend on `node:fs`),
```

Comment:
Would it make sense to mention with stores that are maintained by us don't use the `fs` module.